client/asset/dcr/config.go

```diff
@@ -45,6 +45,9 @@ type Config struct {
 	// Context should be canceled when the application exits. This will cause
 	// some cleanup to be performed during shutdown.
 	Context context.Context `ini:"-"`
+	// AllowExternalAPIRequest is a flag that allows the client to make requests
+	// to external apis.
+	AllowExternalAPIRequest bool `ini:"allowexternalapirequest"`
```

Comment:
`Config` items are set by the user via a corresponding `ConfigOpt` in the `WalletDefinition.ConfigOpts`. I agree that this is a wallet-level configuration option, and this is the right place for this field, but setting the option should not involve **core**, **asset**, or **db** in any way. We already have the system in place to populate `Config`.

```Go
{
	Key:         "allowexternalapirequest",
	DisplayName: "External fee rate estimates",
	Description: "Allow fee rate estimation from a block explorer API. " +
		"This is useful as a fallback for SPV wallets and RPC wallet th" +
		"at have just recently been started.",
	IsBoolean: true,
},
```

Reply:
Removed changes on core, asset and db and left them only on as wallet option. Much simpler indeed.